src/charm.py

```diff
@@ -210,21 +283,108 @@ def _configure(self, event: EventBase) -> None:
         if self._invalid_configs():
             return
         if not self._root_certificate_is_stored or not self._root_certificate_matches_config():
-            self._generate_root_certificate()
             self.tls_certificates.revoke_all_certificates()
+            self._clean_up_juju_secret(EXPIRING_CA_CERTIFICATES_SECRET_LABEL)
+            self._generate_root_certificate()
             logger.info("Revoked all previously issued certificates.")
             return
+        if not self._is_ca_cert_active():
+            logger.info("Renewing CA certificate")
+            self._renew_root_certificate()
+            return
         self._send_ca_cert()
         self._process_outstanding_certificate_requests()
 
+    def _renew_root_certificate(self):
+        """Generate a new active root CA certificate.
+
+        If there is an active CA certificate that is about to expire, generate a new one.
+        If there is no active CA certificate, generate a new one.
```

Comment:
Those if's don't seem to match with the actual lines of code below,

Reply:
Maybe they were not clear enough, I updated them.